### PR TITLE
Drop ChakraCore dependency

### PR DESCRIPTION
**Pull request by JunielKatarn**
ChakraCore is no longer required for any React Native Windows scenarios.
Dropping the direct dependencies as well as administrative code around them.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/9056)

### CODE REVIEW

**Review by NickGerleman — APPROVED** [review]
🙌

**Comment by NikoAri** [comment]
> <?xml version="1.0" encoding="utf-8"?>

Will vnext/Chakra be removed at some point as well? Or split out under a different folder?

---
Refers to: vnext/Chakra/Chakra.vcxitems:1 in 6cbaa07. [](commit_id = 6cbaa07db6e25153e19c47a88259ac2481bd45bb, deletion_comment = False)

File: vnext/Desktop.IntegrationTests/DesktopTestRunner.cpp [hunk 1]
```diff
@@ -90,7 +90,7 @@ shared_ptr<ITestInstance> TestRunner::GetInstance(
   devSettings->platformName = "windows";
 
   // Set to JSIEngineOverride::Chakra when testing the Chakra.dll JSI runtime.
```
Comment (NikoAri) [inline vnext/Desktop.IntegrationTests/DesktopTestRunner.cpp:92]: >Chakra.dll JSI runtime. [](http://example.com/codeflow?start=55&length=23)

Should we switch it to V8? Or abstract it out to make Chakra-specific test?

**Review by NikoAri — APPROVED** [review]
:shipit:

File: vnext/Desktop.IntegrationTests/DesktopTestRunner.cpp (hunk 1 above)
Comment (JunielKatarn) [reply vnext/Desktop.IntegrationTests/DesktopTestRunner.cpp:92]: Sounds good. If it doesn't introduce test failures, I'll switch to V8.

**Comment by JunielKatarn** [comment]
> > 
> 
> Will vnext/Chakra be removed at some point as well? Or split out under a different folder?
> 
> Refers to: vnext/Chakra/Chakra.vcxitems:1 in [6cbaa07](https://github.com/microsoft/react-native-windows/commit/6cbaa07db6e25153e19c47a88259ac2481bd45bb). [](commit_id = [6cbaa07](https://github.com/microsoft/react-native-windows/commit/6cbaa07db6e25153e19c47a88259ac2481bd45bb), deletion_comment = False)

That project is till needed for System Chakra compatibility. I don't know of any plans to remove that dependency in the immediate future.

File: vnext/Desktop.IntegrationTests/DesktopTestRunner.cpp (hunk 1 above)
Comment (dannyvv) [reply vnext/Desktop.IntegrationTests/DesktopTestRunner.cpp:92]: Actually we should probably run it for all supported engine overrides....

File: vnext/Shared/JSI/ChakraRuntime.cpp [hunk 2]
```diff
@@ -16,14 +16,10 @@
 #include <sstream>
 #include <unordered_set>
 
-#ifdef CHAKRACORE
-#include <ChakraCore.h>
-#else
 #ifndef USE_EDGEMODE_JSRT
```
Comment (dannyvv) [inline vnext/Shared/JSI/ChakraRuntime.cpp:19]: Curious if the define of `USE_EDGEMODE_JSRT` has value if this always sets it anyway and we have it defined in all vcxproj files as well.

**Review by dannyvv — APPROVED** [review]

File: vnext/Shared/JSI/ChakraRuntime.cpp (hunk 2 above)
Comment (JunielKatarn) [reply vnext/Shared/JSI/ChakraRuntime.cpp:19]: Probably not. I applied the minimum amount of changes possible until failures (unit tests) are fixed.

Then I'll shrink this code as much as possible.

**Pull request merged**